nbconvert/filters/markdown_mistune.py

```diff
@@ -160,7 +160,10 @@ def image(self, src, title, text):
         attachment_prefix = 'attachment:'
         if src.startswith(attachment_prefix):
             name = src[len(attachment_prefix):]
-            assert name in attachments, "missing attachment: {}".format(name)
+            
+            if not name in attachments:
+                nbconvert.InvalidNotebook.new("missing attachment: {}".format(name))
```

Comment:
```suggestion
                raise InvalidNotebook("missing attachment: {}".format(name))
```

I think this is what you were trying to do.

Reply:
Thank you ❤️